ChiTech/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_01c_finalize.cc

```diff
@@ -3,256 +3,272 @@
-          chi::Exit(EXIT_FAILURE);
+          //create chi_delayed transpose for easier checks
+          EmissionSpectra tmp;
+          tmp.resize(num_precursors, std::vector<double>(num_groups, 0.0));
+          for (unsigned int g = 0; g < num_groups; ++g)
+            for (unsigned int j = 0; j < num_precursors; ++j)
+              tmp[j][g] = chi_delayed[g][j];
+
+          for (unsigned int j = 0; j < num_precursors; ++j)
+          {
+            //throw error if emission spectrum j was not specified
+            if (!was_specified(tmp[j]))
+              throw std::runtime_error(
+                  "The delayed emission spectrum for precursor "
+                  "species " + std::to_string(j) + " was not provided.");
+
+            //normalize each emission spectrum
+            double chidj_sum = std::accumulate(tmp[j].begin(),
+                                               tmp[j].end(), 0.0);
+            for (unsigned int g = 0; g < num_groups; ++g)
+              chi_delayed[g][j] /= chidj_sum;
+          }//for j
         }
-      }
-    }
 
-    //======================================== Normalization
-    if (has_chi)
-    {
-      double chi_sum = 0.0;
-      for (auto& v : chi) chi_sum += v;
-      if (fabs(chi_sum - 1.0) > eps and chi_sum > eps)
-      {
-        chi::log.Log0Warning()
-          << __FUNCTION__ << ": Total fission spectrum does "
-          << "not sum to unity. Normalizing. " << chi_sum-1.0;
-        for (auto& v : chi) v /= chi_sum;
-      }
-    }
+        //check the precursor data
+        {
+          //throw error if decay constants were not specified
+          if (!was_specified(precursor_lambda))
+            throw std::runtime_error(
+                "The delayed neutron precursor decay constants "
+                "was not provided.");
 
-    if (has_chi_p)
-    {
-      double chi_p_sum = 0.0;
-      for (auto& v : chi_prompt) chi_p_sum += v;
-      if (fabs(chi_p_sum - 1.0) > eps and chi_p_sum > eps)
-      {
-        chi::log.Log0Warning()
-            << __FUNCTION__ << ": Prompt fission spectrum does "
-            << "not sum to unity. Normalizing.";
-        for (auto& v : chi_prompt) v /= chi_p_sum;
-      }
-    }
+          //throw error if not specified
+          if (!was_specified(precursor_yield))
+            throw std::runtime_error(
+                "The delayed neutron precursor yields was not provided.");
 
-    if (has_chi_d)
-    {
-      for (int j = 0; j < num_precursors; ++j)
-      {
-        double chi_dj_sum = 0.0;
-        for (int g = 0; g < num_groups; ++g)
-          chi_dj_sum += chi_delayed[g][j];
-        if (fabs(chi_dj_sum - 1.0) > eps and chi_dj_sum > eps)
+          //normalize the precursor yields
+          double yield_sum = std::accumulate(precursor_yield.begin(),
+                                             precursor_yield.end(), 0.0);
+          for (unsigned int j = 0; j < num_precursors; ++j)
+            precursor_yield[j] /= yield_sum;
```

Comment:
If you take a look at how we implement the fission term within the `SetSource` routine, we do not use the formulation $\beta_g^j \nu_g \sigma_{f,g}$ for delayed fission, but rather $\gamma_j \nu^d_{g} \sigma_{f,g}$ where $\gamma_j$ is the fractional yield for precursor species $j$. We just use the definition $\beta_g^j = \gamma_j (\nu^d_g / \nu_g)$. The initial reason for this was because of the way NJOY spits out data.

Reply:
Yeah. We identified this previously. Created an issue #270